astropy/nddata/utils.py

```diff
@@ -243,7 +245,13 @@ def extract_array(array_large, shape, position, mode='partial',
     # Extracting on the edges is presumably a rare case, so treat special here
     if (extracted_array.shape != shape) and (mode == 'partial'):
         extracted_array = np.zeros(shape, dtype=array_large.dtype)
-        extracted_array[:] = fill_value
+        try:
+            extracted_array[:] = fill_value
+        except ValueError:
```

Comment:
I think assuming all `ValueError` is *that* error might be dangerous. Should at least check for the error message before re-raising? Can we guarantee that *only* this kind of error will produce `ValueError` here?

Reply:
I updated the error message to be more generic (e.g. `fill_value='a string'` also raises `ValueError`).  The new error message should cover any `ValueError` raised here because of this assignment.

Reply:
How about keeping the exception chained?  I've also see usage like:
```
except ValueError as exc:
    exc.args += ('Did you perhaps try to insert a float into an integer array',)
    raise exc
```
(But really, we're perhaps going too far with this!!)

Reply:
Hmm `+=` isn't a bad idea. We can see the original error for debugging purpose in the future...